scripts/apps/contacts/helpers.ts

```diff
@@ -75,20 +76,30 @@ export function validateRequiredField(contact) {
     return !invalid;
 }
 
-export function validateMinRequiredField(contact) {
+export function validateMinRequiredField(contact: IContact): boolean {
     return _.some(LOOKUP_FIELDS, (field) => {
         switch (field) {
         case 'contact_email':
-            return _.get(contact, field, 0).length > 0 && !_.some(contact[field], (v) => _.isEmpty(v));
+            return _.get(contact, field, []).length > 0 && !_.some(contact[field], (v) => _.isEmpty(v));
         case 'contact_phone':
         case 'mobile':
-            return _.get(contact, field, 0).length > 0 && !_.some(_.map(contact[field], 'number'), (v) => _.isEmpty(v));
+            return _.get(contact, field, []).length > 0 &&
+                !_.some(_.map(contact[field], 'number'), (v) => _.isEmpty(v));
         default:
             return !_.isEmpty(contact[field]);
         }
     });
 }
 
+export function validateAssignableType(contact: IContact): boolean {
+    if (!_.get(contact, 'contact_type.assignable')) {
+        return true;
+    }
+
+    return _.get(contact, 'contact_email.length', 0) > 0 &&
+        !_.some(contact.contact_email, (v) => _.isEmpty(v));
```

Comment:
Why not use native [Array.prototype.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)?

Is `isEmpty` being used simply as `x != null`? If so, use the native construct instead of a library function.